### PR TITLE
[MI-2121_4]: Test cases for Popover, ResultPanel, SvgWrapper, Tabs, ToggleSwitch and Tooltip components

### PR DESCRIPTION
**Pull request by sooraj-shukla**

### CODE REVIEW

File: src/components/Modal/subComponents/ModalFooter/ModalFooter.test.tsx [hunk 1]
```diff
@@ -21,15 +21,15 @@ describe('Modal Footer', () => {
     let component: ReactWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;
 
     beforeEach(() => {
-        component = mount(<ModalFooter {...modalFooterProps}/>);
+        component = mount(<ModalFooter {...modalFooterProps} />);
```
[the remark this answers is not in the document]
Comment (sooraj-shukla) [reply src/components/Modal/subComponents/ModalFooter/ModalFooter.test.tsx:24]: Fixed.

File: src/components/ResultPanel/ResultPanel.test.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    afterEach(() => {
+        component.unmount();
+    });
+
+    it('Should not render the primary button when "primaryBtn" prop isn\'t passed', () => {
+        expect(component.find('.btn-primary').exists()).toBeFalsy();
+    });
+
+    it('Should not render the secondary button when "secondaryBtn" prop isn\'t passed', () => {
+        expect(component.find('.btn-link').exists()).toBeFalsy();
```
[the remark this answers is not in the document]
Comment (sooraj-shukla) [reply src/components/ResultPanel/ResultPanel.test.tsx:76]: Not required.

File: src/components/Tooltip/Tooltip.test.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        );
+    });
+
+    it('Should render correctly', () => {
+        expect(component).toMatchSnapshot();
+    });
+
+    it('Should render the children correctly', () => {
+        expect(component.contains(TooltipChildren)).toBeTruthy();
+    });
```
[the remark this answers is not in the document]
Comment (sooraj-shukla) [reply src/components/Tooltip/Tooltip.test.tsx:33]: We are providing the `className` directly to the `ReactTooltip` component which is imported from the `react-bootstrap` and the library has its own tests, so can't we rely on it?

File: src/components/Card/Subscription/SubscriptionCard.test.tsx [hunk 4]
```diff
@@ -1,68 +1,10 @@
...
-    list: ['List Item 1', 'List Item 2', 'List Item 3'],
-    labelValuePairs: [
-        {label: 'Label 1', value: 'Value 1'},
-        {label: 'Label 2', value: 'Value 2'},
-        {label: 'Label 3', value: 'Value 3'},
-        {label: 'Label 4', value: 'Value 4'},
-    ],
-};
+// Mock data
+import { subscriptionCardBody as cardBody, subscriptionCardProps } from '@Constants/mockData/card';
```
Comment (manojmalik20) [inline src/components/Card/Subscription/SubscriptionCard.test.tsx:5]:
```suggestion
import {subscriptionCardBody as cardBody, subscriptionCardProps} from '@Constants/mockData/card';
```
There should also be an eslint rule for this.

File: src/components/Dropdown/Dropdown.test.tsx [hunk 5]
```diff
@@ -1,23 +1,10 @@
 import React from 'react';
 import {mount, ReactWrapper} from 'enzyme';
 
-import Dropdown, {DropdownProps} from '.';
-
-// Mock functions
-const onChangeHandler = jest.fn();
+// Mock data
+import { onChangeHandler, dropdownOptions, dropdownPlaceholder } from '@Constants/mockData/dropdown';
```
Comment (manojmalik20) [inline src/components/Dropdown/Dropdown.test.tsx:5]:
```suggestion
import {onChangeHandler, dropdownOptions, dropdownPlaceholder} from '@Constants/mockData/dropdown';
```

File: src/components/Modal/subComponents/ModalLoader/__snapshots__/ModalLoader.test.tsx.snap [hunk 6]
```diff
@@ -2,4 +2,4 @@
 
 exports[`Modal Body Should render correctly when loading is false 1`] = `ReactWrapper {}`;
 
-exports[`Modal Loader - loading state Should render correctly when loading is true 1`] = `ShallowWrapper {}`;
+exports[`Modal Loader - loading state Should render correctly when "loading" is true 1`] = `ShallowWrapper {}`;
```
Comment (manojmalik20) [inline src/components/Modal/subComponents/ModalLoader/__snapshots__/ModalLoader.test.tsx.snap:5]: Still not correct.

File: src/constants/index.ts [hunk 7]
```diff
@@ -6,3 +6,5 @@ export enum ToggleSwitchLabelPositioningValues {
 }
 
 export const TooltipDefaultPlacement = 'top';
+
+export const PrivateChannelType = 'P';
```
Comment (manojmalik20) [inline src/constants/index.ts:10]: Why did you add this again and not using `mattermost-redux`?

File: src/constants/mockData/resultPanel.tsx [hunk 8]
```diff
@@ -0,0 +1,18 @@
+// Mock data for "ResultPanel" component
+export const primaryBtnHandler = jest.fn();
+export const secondaryBtnHandler = jest.fn();
```
Comment (manojmalik20) [inline src/constants/mockData/resultPanel.tsx:3]: Fix the extension of all the files. If there is no need of `.tsx` extension, then convert it to `.ts`.

File: src/components/Card/Subscription/SubscriptionCard.test.tsx (hunk 4 above)
Comment (sooraj-shukla) [reply src/components/Card/Subscription/SubscriptionCard.test.tsx:5]: This was a lint error, there is already a rule for this.

File: src/constants/index.ts (hunk 7 above)
Comment (sooraj-shukla) [reply src/constants/index.ts:10]: Added to-do to use `mattermost-redux` as discussed.

**Review by manojmalik20 — APPROVED** [review]
